src/storage/FileLoader.cpp

```diff
@@ -38,17 +38,19 @@ std::string FileLoader::getHashFilePath(const std::string& path)
 }
 
 std::vector<uint8_t> FileLoader::doCodegen(std::vector<uint8_t>& bytes,
-                                           const std::string& fileName)
+                                           const std::string& fileName,
+                                           bool isSgx)
 {
     conf::FaasmConfig& conf = conf::getFaasmConfig();
     if (conf.wasmVm == "wamr") {
 #if (WAMR_EXECTION_MODE_INTERP)
         throw std::runtime_error(
           "WAMR codegen not supported with WAMR interp mode");
 #else
-        return wasm::wamrCodegen(bytes);
+        return wasm::wamrCodegen(bytes, isSgx);
 #endif
     } else {
+        assert(isSgx == false);
```

Comment:
We already check this in the same file (lines 74-76) so I thought an assertion would suffice here.